components/TopUpXcmFee.tsx

```diff
@@ -0,0 +1,216 @@
+
+  const [switchPair, setSwitchPair] = useState<SwitchPair | null>(null);
+  const [xcmFee, setXcmFee] = useState<string | null>(null);
+
+  // Getting the correct XCM fee and switch pair based on the props.
+  useEffect(() => {
+    const { switchPair: switchPair1 } =
+      source.id === "assethub"
+        ? parachainConfigs[destination.name]
+        : parachainConfigs[source.name];
+
+    const xcmFee1 = getFormattedBalance(
+      BigInt(switchPair1[0].xcmFee.amount),
+      switchPair1[0].xcmFee.decimals,
+    );
+    setSwitchPair(switchPair1);
+    setXcmFee(xcmFee1);
+  }, [source.id, source.name, destination.name]);
+
+  const [busyMessage, setBusyMessage] = useState("");
+
+  const [amountInput, setAmountInput] = useState("");
+
+  const [error, setError] = useState<ErrorInfo | null>(null);
+
+  const submitTopUp = useCallback(async () => {
+    if (!context || !switchPair || !xcmFee) return;
+
+    if (source.name === destination.name) return;
+
+    if (amountInput > xcmFee) return;
+
+    try {
+      if (!sufficientTokenAvailable) {
+        setError({
+          title: "Not Enough Sufficient tokens",
+          description: "Please follow the sufficient or existential deposit",
+          errors: [
+            {
+              kind: "toPolkadot",
+              code: toPolkadot.SendValidationCode.BeneficiaryAccountMissing,
+              message:
+                "Asset Hub requires that you hold specific tokens in order for an account to be active.",
+            },
+          ],
+        });
+      }
+
+      setBusyMessage("Transaction being created.");
+      const api = context.polkadot.api.assetHub;
+
+      const tx = api.tx.polkadotXcm.limitedReserveTransferAssets(
+        {
+          V4: {
+            parents: 1,
+            interior: {
+              X1: [{ Parachain: source.paraInfo?.paraId! }],
+            },
+          },
+        },
+        {
+          V4: {
+            parents: 0,
+            interior: {
+              X1: [
+                {
+                  AccountId32: {
+                    id: decodeAddress(beneficiary),
+                  },
+                },
+              ],
+            },
+          },
+        },
+        {
+          V4: [
+            {
+              id: switchPair[0].xcmFee.remoteXcmFee.V4.id,
+              fun: {
+                Fungible: parseUnits(
+                  amountInput,
+                  switchPair[0].xcmFee.decimals,
+                ),
+              },
+            },
+          ],
+        },
+        0,
+        "Unlimited",
+      );
+
+      const { signer, address } = polkadotAccounts?.find(
+        (val: WalletAccount) => val.address === sourceAccount,
+      )!;
+
+      if (!signer) {
+        throw new Error("Signer is not available");
+      }
+
+      setBusyMessage("Transaction in flight.");
+
+      await tx.signAndSend(address, { signer }, (result) => {
+        result.isFinalized;
+      });
```

Comment:
either handle transaction status updates and set busy/error messages accordingly, or just change to  
```suggestion
      await tx.signAndSend(address, { signer });
```

Reply:
Updated to handle this and set the busy and results toast

https://github.com/Snowfork/snowbridge-app/pull/23/commits/9cd1da27c669c03186fe6a87466519c94f98469a